File: gc-starter/crud-spring-boot-starter/src/main/java/com/gc/starter/crud/model/PageData.java
package com.gc.starter.crud.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.util.List;

/**
 * 分页数据
 * @author shizhongming
 * 2020/1/12 8:30 下午
 */
@AllArgsConstructor
@Getter
@Setter
public final class PageData<T extends Serializable> implements Serializable {

    private static final long serialVersionUID = 3546130315339330080L;

    private List<T> rows;

    private Long total = 0L;
}

File: smart-starter/crud-spring-boot-starter/src/main/kotlin/com/smart/starter/crud/model/PageData.kt
package com.smart.starter.crud.model

import java.io.Serializable

/**
 * 分页数据
 * @author ming
 * 2019/9/13 下午7:24
 */
class PageData<T: BaseModel>(val rows: List<T>?, val total: Long = 0) : Serializable {
}